src/provider/application.js

```diff
@@ -260,10 +260,18 @@ class Application extends EventEmitter {
   }
 
   unload() {
+    const { activeElement } = document;
+
+    const hasDownload = activeElement ? activeElement.hasAttribute('download') : false;
+    const hrefValue = activeElement && activeElement.hasAttribute('href') ? activeElement.getAttribute('href') : '';
+
+    // Need to check for href attributes to avoid triggering unload.
+    const attributeCheck_href = hrefValue.includes('tel:') || hrefValue.includes('mailto:') || hrefValue.includes('fax:') ||
+     hrefValue.includes('sms:') || hrefValue.includes('callto:');
+
     // Need this line because IE11 & some safari trigger onbeforeunload despite presence of download attribute
-    if (document.activeElement && document.activeElement.hasAttribute('download')) {
-      return;
-    }
+    if (hasDownload || attributeCheck_href) return;
```

Comment:
The logic can be simplified a bit using regex and wrapping the logic in the conditional rather than returning early.

```js
// These patterns trigger unload events but don't actually unload the page
const protocols = /^(tel|mailto|fax|sms|callto):/;
const element = document.activeElement;

if (!element || !(element.hasAttribute('download') || protocols.test(element.href))) { 
    this.JSONRPC.notification('unload');
    this.trigger('xfc.unload');
}
```

Reply:
I agree with the `protocols` regex – though I _personally_ don't like if-statement wrapper guards, though this is just a matter of taste/style. I'd rather read it like a ruby `unless` statement, and then perform logic below the guard so it appears to be a logical part of the functions purpse for existense, rather than some conditional logic.

_Yes,_ this one is **incredibly** small and **incredibly** simple, but I _personally_ prefer to keep things consistent across the board. (I also think @slor prefers early-returns as well ¯\_(ツ)_/¯

Reply:
In this case early return is not required, as we are checking for the existence of element and the other attributes at the same level, so even if we do

```js
if(!element) return;

if(!(element.hasAttribute('download') || protocols.test(element.href))) {
   this.JSONRPC.notification('unload');
   this.trigger('xfc.unload');
}
```
it doesn't make much difference.